thefuck/rules/omnienv_no_such_command.py

```diff
@@ -1,33 +1,34 @@
 import re
-from subprocess import PIPE, Popen
-
 from thefuck.utils import (cache, for_app, replace_argument, replace_command,
                            which)
+from subprocess import PIPE, Popen
+
+
+supported_apps = 'goenv', 'nodenv', 'pyenv', 'rbenv'
+enabled_by_default = any(which(a) for a in supported_apps)
+
 
 COMMON_TYPOS = {
     'list': ['versions', 'install --list'],
     'remove': ['uninstall'],
 }
 
 
-@for_app('pyenv')
+@for_app(*supported_apps, at_least=1)
 def match(command):
-    return 'pyenv: no such command' in command.output
+    return 'env: no such command ' in command.output
 
 
-def get_pyenv_commands():
-    proc = Popen(['pyenv', 'commands'], stdout=PIPE)
+def get_app_commands(app):
+    proc = Popen([app, 'commands'], stdout=PIPE)
     return [line.decode('utf-8').strip() for line in proc.stdout.readlines()]
 
 
-if which('pyenv'):
-    get_pyenv_commands = cache(which('pyenv'))(get_pyenv_commands)
-
-
-@for_app('pyenv')
 def get_new_command(command):
-    broken = re.findall(r"pyenv: no such command `([^']*)'", command.output)[0]
+    app = command.script_parts[0]
+    app_commands = cache(which(app))(get_app_commands)(app)
+    broken = re.findall(r"env: no such command ['`]([^']*)'", command.output)[0]
     matched = [replace_argument(command.script, broken, common_typo)
                for common_typo in COMMON_TYPOS.get(broken, [])]
```

Comment:
Separating out the two blocks of matching logic into different code blocks should be more readable.

```suggestion
    matched = [replace_argument(command.script, broken, common_typo)
               for common_typo in COMMON_TYPOS.get(broken, [])]

    app = command.script_parts[0]
    app_commands = cache(which(app))(get_app_commands)(app)
    broken = re.findall(r"env: no such command ['`]([^']*)'", command.output)[0]
```

Reply:
👍